frontend/src/pages/DetailedCourse/DetailedCourse.tsx

```diff
@@ -55,9 +62,20 @@ const DetailedCourse: React.FC<IProps> = ({ handleApplyCourse }) => (
               lessons={INITIAL_DETAILED_COURSE.lessons}
             />
           </CourseInfoBox>
-          <StartButton variant="large" color="primary" onClick={handleApplyCourse}>
-            Start
-          </StartButton>
+          {status === 'loading' && targetId === buttonId.start ? (
```

Comment:
!
In case you will make any kind of mistake in string, this condition will crash logic of the page!
This case breaks all the practice paradigms

Reply:
Replaced on isLoading